### PR TITLE
Add 'filter:middleware.renderHeader' hook

### PR DESCRIPTION
**Pull request by jsdream**
Alters data passed to header template.

### CODE REVIEW

**Comment by jsdream** [comment]
Hello!)
Any chances this will be merged?

**Comment by barisusakli** [comment]
@jsdream I see no problems, what are you using this for?

File: src/middleware/middleware.js [hunk 1]
```diff
@@ -286,7 +286,9 @@ middleware.renderHeader = function(req, res, callback) {
 		templateValues.template = {name: res.locals.template};
 		templateValues.template[res.locals.template] = true;
 
-		app.render('header', templateValues, callback);
+		plugins.fireHook('filter:middleware.renderHeader', {templateValues: templateValues, req: req, res: res}, function(err, data) {
+            		app.render('header', data.templateValues, callback);
```
Comment (barisusakli) [inline src/middleware/middleware.js]: please add a err check here

```
if (err) {
    return callback(err);
} 
 ```

**Comment by jsdream** [comment]
I'm using this hook to expose custom user fields to header template. I've added error check to pull request.

**Comment by barisusakli** [comment]
@jsdream have you signed https://docs.google.com/forms/d/1KrniXs3pwK8dMeAZnnQz2D07ne7LjJTYIYwJrpDqgnE/viewform

**Comment by jsdream** [comment]
Yes, I did that. This is my second pull request to NodeBB actually.

**Comment by barisusakli** [comment]
Thanks @jsdream :smile:

**Comment by jsdream** [comment]
You too. You guys doing very nice product ;-)

**Pull request merged**